image/v1/types.go

```diff
@@ -428,9 +428,14 @@ type ImageStreamLayers struct {
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 	// blobs is a map of blob name to metadata about the blob.
 	Blobs map[string]ImageLayerData `json:"blobs" protobuf:"bytes,2,rep,name=blobs"`
-	// images is a map between an image name and the names of the blobs and manifests that
+	// images is a map between an image name and the names of the blobs and config that
 	// comprise the image.
 	Images map[string]ImageBlobReferences `json:"images" protobuf:"bytes,3,rep,name=images"`
+	// imagesWithoutMetadata contains information about all images referenced by an image
+	// stream whose metadata could not be loaded. This can be due to an administrator or
+	// other automated process deleting the image (in which case the image cannot be pulled)
+	// or be a temporary condition because the image has not yet been uploaded.
```

Comment:
@smarterclayton when can we have a reference to the image that has not yet been uploaded?

Reply:
If someone writes an integration that uses the API but does so without verifying the references.  We don't allow it directly, but I want to leave this open so that a consumer of the API knows that this is possible even if we're confident none of our existing code paths allow it.

Reply:
I can change the godoc if you'd like, but I do want to call out that this API is dependent on both the object references and the object metadata.

Reply:
is it better to have this as a separate field, or have a `missing` attribute of `ImageBlobReferences` that can only be true if `config` and `layers` are unset?

Reply:
I could do either.

Reply:
@deads2k opinion on ^?

Reply:
@liggitt @smarterclayton I prefer a missing attribute on ImageBlobReferences.  I don't care for parallel maps.

Reply:
ok